vc-verifier/js/src/utils/CredentialUtils.js

```diff
@@ -0,0 +1,16 @@
+import { CredentialVerifierConstants } from "../constant/CredentialVerifierConstants";
+
+export const preProcessVerifiableCredential = (credential) => {
+    if (!credential["proof"]) {
+        throw new Error("Proof document is not available in the received credentials.");
+    }
+    const jsonLdProof = credential["proof"];
+    if (!jsonLdProof?.["type"] || jsonLdProof["type"] !== CredentialVerifierConstants.SIGNATURE_SUITE_TERM) {
+        throw new Error("Proof Type is not supported. Recevied Type: " + jsonLdProof?.["type"]);
+    }
+    if (!credential["type"].includes('VerifiableCredential')) {
+        throw new Error("Credential is not of type Verifiable Credential.")
+    }
+    credential['type'] = 'VerifiableCredential'
```

Comment:
why are we overriding type attribute?

Reply:
Because only for Verifiable credential type wwe have the context available and it is able to do the mapping. For "MOSIPVerifiableCredential" or any other credential type it is failing., Web of trust library is intelligent enough to remove it at run time. jsonLd library is not doing it. So we have to remove it manually.

Reply:
can you point me to web of trust library reference for this?

Reply:
Had a discussion on call, resolving it